static/js/app.ide.js

```diff
@@ -134,27 +153,28 @@ export default class IDEApp extends App {
    */
   async setEditorFileContent(editorComponent, clearUndoStack = false) {
     const filepath = editorComponent.getPath();
+    if (!filepath) return;
 
-    if (!filepath || (filepath && !(await this.vfs.pathExists(filepath)))) {
-      return;
-    }
-
-    // Check if the editor exceeded the maximum allowed file size.
-    const size = await this.vfs.getFileSizeByPath(filepath);
-    if (size > MAX_FILE_SIZE) {
-      editorComponent.exceededFileSize();
-      return;
-    }
+    try {
+      const content = await this.vfs.readFile(filepath, { MAX_FILE_SIZE });
 
-    const content = await this.vfs.getFileContentByPath(filepath);
-    editorComponent.setContent(content);
+      editorComponent.setContent(content);
 
-    const cursorPos = editorComponent.getCursorPosition();
-    editorComponent.setContent(content);
-    editorComponent.setCursorPosition(cursorPos);
+      const cursorPos = editorComponent.getCursorPosition();
+      editorComponent.setContent(content);
+      editorComponent.setCursorPosition(cursorPos);
 
-    if (clearUndoStack) {
-      editorComponent.clearUndoStack();
+      if (clearUndoStack) {
+        editorComponent.clearUndoStack();
+      }
+    } catch (err) {
+      if (err instanceof FileTooLargeError) {
```

Comment:
Errors in een worker throwen kan je niet opvangen, dus dit heeft geen nut hier. Ik heb dit nog even dubbel gecheckt op je branch.

Daarom gebruik ik in andere workers callback functies en vang deze op in de bridge class tussen de app en de worker om te bepalen wat ik doe als er een error vanuit de worker komt.